src/Thentos/Transaction/Transactions.hs

```diff
@@ -49,6 +49,10 @@ assertUser mUid user = polyQuery $ ask >>= \ db ->
        | userFacetExists (^. userEmail) mUid user db -> throwT UserEmailAlreadyExists
        | True -> return ()
 
+-- | Assert that no user with the same name or email address already exists in the DB.
+trans_assertUserIsNew :: (AsDB db) => User -> ThentosQuery db ()
+trans_assertUserIsNew = assertUser Nothing
```

Comment:
@ChristianSi has pointed out that this has nothing to do with `UserId`, but checks freshness of name and email.  Please ignore previous comment.